cmd/print_test.go

```diff
@@ -58,3 +75,146 @@ func Test_getResourcePrinter(t *testing.T) {
+		expectedNamespaceFilter   string
+		expectingError            bool
+		expectingCurrentNamespace bool
+	}{
+		{
+			name:                      "Only namespace is specified",
+			namespace:                 "default",
+			allNamespaces:             false,
+			expectedNamespaceFilter:   "default",
+			expectingError:            false,
+			expectingCurrentNamespace: false,
+		},
+		{
+			name:                      "All namespaces overrides a specific namespace",
+			namespace:                 "default",
+			allNamespaces:             true,
+			expectedNamespaceFilter:   "",
+			expectingError:            false,
+			expectingCurrentNamespace: false,
+		},
+		{
+			name:                      "Current namespace used when nothing specified",
+			namespace:                 "",
+			allNamespaces:             false,
+			expectedNamespaceFilter:   "_",
+			expectingError:            false,
+			expectingCurrentNamespace: true,
+		},
+	}
+
+	destroy, err := setupKubeConfig()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer destroy()
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actualNamespaceFilter, err := getNamespaceFilter(tc.namespace, tc.allNamespaces)
+
+			if tc.expectingError && err == nil {
+				t.Errorf("Expected error but got none")
+			}
+			if !tc.expectingError && err != nil {
+				t.Errorf("Expected no error but got %v", err)
+			}
+
+			if tc.expectingCurrentNamespace {
+				tc.expectedNamespaceFilter, _ = getNamespaceFilter(tc.namespace, tc.allNamespaces)
+			}
+
+			if actualNamespaceFilter != tc.expectedNamespaceFilter {
+				t.Errorf(`getNamespaceFilter("%s", %v) = %v, expected %v`,
+					tc.namespace, tc.allNamespaces, actualNamespaceFilter, tc.expectedNamespaceFilter)
+			}
+		})
+
+	}
+}
+
+func setupKubeConfig() (func(), error) {
+	const kubeConfigPath = "/tmp/i2gw/.kube"
+	kubeConfigFile := fmt.Sprintf("%s/config", kubeConfigPath)
+
+	if err := os.Setenv("KUBECONFIG", kubeConfigFile); err != nil {
+		return nil, err
+	}
+
+	// Clean up from the last test, just in case...
+	os.Remove(kubeConfigFile)
+
+	content := []byte(`
+apiVersion: v1
+clusters:
+- cluster:
+    server: https://kubernetes.docker.internal:6443
+  name: docker-desktop
+- cluster:
+    server: https://127.0.0.1:54873
+  name: kind-i2gw
+contexts:
+- context:
+    cluster: docker-desktop
+    namespace: non-default-ns
+    user: docker-desktop
+  name: docker-desktop
+- context:
+    cluster: kind-i2gw
+    user: kind-i2gw
+  name: kind-i2gw
+current-context: docker-desktop
+kind: Config
+preferences: {}
+`)
+
+	err := os.MkdirAll(kubeConfigPath, os.ModePerm)
+	if err != nil {
+		log.Println(err)
+	}
+
+	f, err := os.Create(kubeConfigFile)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err = f.Write(content); err != nil {
+		os.Remove(kubeConfigFile)
+		return nil, err
+	}
+	if err = f.Close(); err != nil {
+		os.Remove(kubeConfigFile)
+		return nil, err
+	}
+
+	return func() {
+		os.Remove(kubeConfigFile)
+	}, nil
+}
```

Comment:
This feels strange - I don't think we want our tests to depend on docker desktop. Can we just use kubernetes utils for getting kube config? I'm probably missing some context here.

Reply:
The test doesn't depend on docker desktop (or any other provider). I just wanted to configure a custom kube config file in order to test that getNamespaceInCurrentContext() function can successfully extract the namespace out of it.

I didn't want to assume the user running tests already has a file, so I decided to create a new temporary one with non-default values that I can work with. Doesn't matter if the clusters are from Docker, AKS, EKS, or anything else since the format of kube config stays the same.

Reply:
That makes sense, maybe replace the references to "docker-desktop" with something like "example" or "test" so the purpose of this is clear.

Reply:
sure, changed it